### PR TITLE
Add the first iteration of Giphy Picker

### PR DESCRIPTION
**Pull request by shiki**
Giphy Picker

This is the first iteration for supporting Giphy (wordpress-mobile/WordPress-Android#8627). Only searching and endless scroll are currently available. For now, the picker can only be launched from the Media Browser.

The search uses the [Paging Library](https://goo.gl/oNfytZ) from Android Jetpack. The process works like this:

0. `GiphyPickerActivity`: The `RecyclerView` is wired to receive events from `GiphyPickerViewModel.mediaViewModelPagedList` and is updated using `GiphyPickerPagedListAdapter`
1. `GiphyPickerActivity`: User enters search text and presses _Done_
2. `GiphyPickerViewModel` notifies `GiphyPickerDataSourceFactory` that there's a new search text
3. `GiphyPickerDataSourceFactory` invalidates the `GiphyPickerDataSource` and creates a new instance of it
4. `GiphyPickerDataSource` executes an initial Giphy API request. This initial load sets up the total count of GIFs that we can expect from the API.
5. `GiphyPickerActivity`: The `RecyclerView` will be notified when the request is done and updates the list

The endless scroll is also automatically handled by the `GiphyPickerPagedListAdapter`.

## UI

The layout files from `StockMediaPickerActivity` are reused since they are very similar. These are `stock_media_picker_thumbnail` and `stock_media_picker_activity`.

## Testing

1. Create a [Giphy API key](https://developers.giphy.com/docs/) and paste it in your `gradle.properties` as the value for `wp.giphy.api_key`
2. Go to the Media Browser
3. Click on the _Add_ button at the top right. Select the _Choose from Giphy_ option
4. In the Giphy picker, type what you want to search for
5. Press Enter on the keyboard to search

## Known Issues

- If you test this long enough, you will eventually hit API rate limits. This will cause empty blocks in the grid.
- Error handling for network errors will be added later

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/giphy/GiphyPickerActivity.kt [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        (application as WordPress).component().inject(this)
+
+        viewModel = ViewModelProviders.of(this).get(GiphyPickerViewModel::class.java)
+
+        // We are intentionally reusing this layout since the UI is very similar.
+        setContentView(R.layout.stock_media_picker_activity)
```
Comment (kwonye) [inline WordPress/src/main/java/org/wordpress/android/ui/giphy/GiphyPickerActivity.kt]: Maybe we should rename this to make sense between both activities?

Comment (shiki) [reply WordPress/src/main/java/org/wordpress/android/ui/giphy/GiphyPickerActivity.kt]: Good idea. What do you think about simply removing the `stock_`?

- `media_picker_activity`
- `media_picker_thumbnail`

Comment (kwonye) [reply WordPress/src/main/java/org/wordpress/android/ui/giphy/GiphyPickerActivity.kt]: Yea, let's do that! Considering these are generic that's perfect.

**Comment by kwonye** [comment]
This looks and works great! Definitely able to merge to the master branch 👍 

:shipit:

**Comment by kwonye** [comment]
(once you change the naming)

**Pull request merged**